### PR TITLE
TL/MLX5: fix warning and var names

### PR DESCRIPTION
**Pull request by MamziB**
TL/MLX5: fix warning and var names
checking if the mcast ctx is created, then we attempt to create the mcast team to avoid warning + some of the variable names got enhanced

### CODE REVIEW

File: src/components/tl/mlx5/mcast/tl_mlx5_mcast_team.c [hunk 1]
```diff
@@ -62,7 +62,7 @@ ucc_status_t ucc_tl_mlx5_mcast_team_init(ucc_base_context_t *base_context,
     ucc_tl_mlx5_mcast_coll_comm_t           *comm;
     int                                      i;
 
-    if (!ctx->mcast_enabled || !ctx->mcast_ready || NULL == mcast_context) {
+    if (!ctx->mcast_enabled || !ctx->mcast_ctx_ready || NULL == mcast_context) {
```
[the remark this answers is not in the document]
Comment (MamziB) [reply src/components/tl/mlx5/mcast/tl_mlx5_mcast_team.c:65]: Thanks for the note, I removed the unnecessary checks. 
_ctx->mcast_ctx_ready has already been checked before calling the function_ I would like to still keep this check, to make sure this function works even if we do not check mcast context before calling it.

File: src/components/tl/mlx5/tl_mlx5_team.c [hunk 2]
```diff
@@ -188,7 +191,7 @@ ucc_status_t ucc_tl_mlx5_team_create_test(ucc_base_team_t *team)
                 tl_debug(UCC_TL_TEAM_LIB(tl_team),
                          "failure during mcast ctx create, no mcast team support");
```
[the remark this answers is not in the document]
Comment (MamziB) [reply src/components/tl/mlx5/tl_mlx5_team.c:192]: I enhanced it in the new commit to "**some of the ranks do not have mcast context available so no mcast team is created**"

File: src/components/tl/mlx5/tl_mlx5_team.c [hunk 3]
```diff
@@ -230,14 +233,14 @@ ucc_status_t ucc_tl_mlx5_team_create_test(ucc_base_team_t *team)
...
-            tl_debug(team->context->lib, "attempted to initialize tl team: %p: MCAST component is %s ALLTOALL component is %s",
+            tl_debug(team->context->lib, "team %p: MCAST component is %s ALLTOALL component is %s",
                     team, (tl_team->mcast_state == TL_MLX5_TEAM_STATE_MCAST_READY)?"ENABLED":"DISABLED",
                     (tl_team->a2a_state == TL_MLX5_TEAM_STATE_ALLTOALL_READY)?"ENABLED":"DISABLED");
         }
 
         if (tl_team->mcast_state == TL_MLX5_TEAM_STATE_MCAST_NOT_AVAILABLE &&
             tl_team->a2a_state == TL_MLX5_TEAM_STATE_ALLTOALL_NOT_AVAILABLE) {
-            tl_warn(team->context->lib, "unable to initialize tl team as both ALLTOALL and MCAST are not available: %p", team);
+            /* both of a2a and mcast components are not available so no need to create the mlx5 team  */
```
[the remark this answers is not in the document]
Comment (MamziB) [reply src/components/tl/mlx5/tl_mlx5_team.c:243]: Nice point here. I changed it. In the new commit, I am returning UCC_OK even if a2a and mcast are not available. During team cleanup we already check if these two components are available then we perform the clean up for them.

**Comment by MamziB** [comment]
Hi @samnordmann Thanks for the comments, I have addressed all of them in the new commit. Can you please take a look and press the Resolve button yourself?

**Review by samnordmann — APPROVED** [review]

**Comment by MamziB** [comment]
@samnordmann oh sorry, I re-requested to review by mistake.

**Review by samnordmann — APPROVED** [review]

File: src/components/tl/mlx5/mcast/tl_mlx5_mcast_team.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (samnordmann) [reply src/components/tl/mlx5/mcast/tl_mlx5_mcast_team.c]: in this case an "assert" is more relevant imo, but I leave it up to you

**Review by Sergei-Lebedev — APPROVED** [review]

**Pull request merged**